python/pyarrow/tests/test_compute.py

```diff
@@ -1212,6 +1212,80 @@ def test_strptime():
     assert got == expected
 
 
+def _check_datetime_components(ts):
+    tsa = [pa.array(ts)]
+    subseconds = ((ts.microsecond * 10**3 + ts.nanosecond) *
+                  10**-9).values.round(9)
+    iso_calendar_fields = [
+        pa.field('iso_year', pa.int64()),
+        pa.field('iso_week', pa.int64()),
+        pa.field('iso_day_of_week', pa.int64())
+    ]
+    iso_calendar = pa.StructArray.from_arrays([
+        ts.isocalendar()["year"].astype(int),
+        ts.isocalendar()["week"].astype(int),
+        ts.isocalendar()["day"].astype(int)],
+        fields=iso_calendar_fields)
+
+    assert pc.call_function("year", tsa).equals(pa.array(ts.year))
```

Comment:
You can directly use `pc.year(..)` instead of `pc.call_function("year", ..)` (and the same for the others below)

Reply:
Done.